src/main/java/bb/example/UserInterface.java

```diff
@@ -1,49 +1,67 @@
 package bb.example;
 
-import java.io.BufferedWriter;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.nio.file.StandardOpenOption;
+import bb.example.services.IOHandler.InputHandler;
+import bb.example.services.IOHandler.OutputHandler;
+import bb.example.services.Utilities;
+import bb.example.services.Validators;
+
 import java.util.Collection;
-import java.util.Scanner;
 
 /**
  * The class interacts with the user.
  *
  * @author pollib
  */
 public class UserInterface {
-    private final Scanner scanner = new Scanner(System.in);
     private final PersonDatabase database;
+    private final InputHandler inputHandler;
+    private final OutputHandler outputHandler;
 
-    public UserInterface(PersonDatabase database) {
+    public UserInterface(PersonDatabase database, InputHandler inputHandler, OutputHandler outputHandler) {
         this.database = database;
+        this.inputHandler = inputHandler;
+        this.outputHandler = outputHandler;
     }
 
-    public String menuSelection(String inputOption) {
-
-        int inputSelection = enterNumber(inputOption);
-        String result = "";
-
-        switch (inputSelection) {
-            case 1 -> result = addInsured();
-            case 2 -> result = displayAllInsured();
-            case 3 -> result = displayInsured();
-            case 4 -> result = modifyInsured();
-            case 5 -> result = deleteInsured();
-            case 6 -> result = createFile();
-            case 7 -> result = "Goodbye";
-            default -> result = "Enter number from 1 to 7";
+    public void menuSelectionLoop(InputHandler inputHandler) {
+        while (true) {
+            System.out.println("""
```

Comment:
This should probably use OutputHandler as well

Reply:
In fact there are more usages of `System.out`, they all should be replaced by `outputHandler.print()`